scripts/west_commands/completion/west-completion.fish

```diff
@@ -196,11 +196,21 @@ function __zephyr_west_complete_help
 end
 
 function __zephyr_west_complete_board
+    # HWMv1
     set -l boards (west 2>/dev/null boards --format="{name} {arch}")
     for board in $boards
         set -l b (string split " " $board)
         printf "%s\n" $b[1]\t"$b[2]"
     end
+
+    # HWMv2
+    set -l boards (west 2>/dev/null boards --format="{identifiers}")
+    for board in $boards
+        set -l b (string split "," $board)
+        for variant in $b
+            printf "%s\n" $variant[1]
```

Comment:
Right now this will print `<board>/<identifier>    (board to build for)`
Ref: https://github.com/zephyrproject-rtos/zephyr/blob/69b0952ff1d676148761ac19ca059596d7b48a86/scripts/west_commands/completion/west-completion.fish#L310

Should we consider writing something else, perhaps vendor ?
```
            printf "%s\n" $variant[1]\t"<vendor>"
```

Could be done by supporting `west boards --format="{identifiers} {vendor}"`
In HWMv2, vendor is present in `board.yml`, and thus could be added to format printing.

If this sounds like a valid use-case, then you can create an enhancement issue.

That could give:
```
$ west build -b nrf52<TAB>
nrf52832_mdk/nrf52832  (Nordic Semiconductor)  nrf52840dk/nrf52811      (Nordic Semiconductor) ...